### PR TITLE
alias: add new backend to create aliases for remote names

### PR DESCRIPTION
**Pull request by B4dM4n**
The alias backend is just a wrapper for an existing remote.
The `Fs` returned for `myalias:subfolder` is the same as using `cache:root/subfolder` with the following config:

```ini
[myalias]
type = alias
remote = cache:root
```

This is a proposal for #1049
There is currently no documentation. I will write it once the backend is accepted.

### CODE REVIEW

File: backend/alias/alias.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// Register with Fs
+func init() {
+	fsi := &fs.RegInfo{
+		Name:        "alias",
+		Description: "Alias for a existing remote",
+		NewFs:       NewFs,
+		Options: []fs.Option{{
+			Name: "remote",
+			Help: "Remote to alias.\nNormally should contain a ':' and a path, eg \"myremote:path/to/dir\",\n\"myremote:bucket\" or maybe \"myremote:\".",
```
Comment (ncw) [inline backend/alias/alias.go]: Looking at the code I think it should allow just putting a path in, so `/path/to/folder` could be an option here too.

Comment (B4dM4n) [reply backend/alias/alias.go]: It's already possible to enter a local path. 
The option is mainly copied from the crypt backend. 
It is also possible specify a local path there (i'm pretty sure).

I will change the documentation with the next commit.

Comment (ncw) [reply backend/alias/alias.go]: I probably didn't put that docs in the crypt as I didn't want to confuse people, whereas I think making local aliases will be an important use of this :-)

**Comment by ncw** [comment]
I think this is a nice idea - please carry on :-)

**Comment by B4dM4n** [comment]
I think the `alias` remote is now in a good shape. The docs have been updated and i specified the handling of `..` in paths.

@ncw Do you think i should add some tests? The only thing that could be tested is the path handling during `NewFS` and that is the same as in `fs.ParseRemote`.

**Comment by ncw** [comment]
> I think the alias remote is now in a good shape. The docs have been updated and i specified the handling of .. in paths.

I agree - it looks good :-)

> Do you think i should add some tests? The only thing that could be tested is the path handling during NewFS and that is the same as in fs.ParseRemote.

The least effort thing to do would be to add the alias remote to [gen_tests.go](https://github.com/ncw/rclone/blob/master/fstest/fstests/gen_tests.go#L169), run `make gen_tests` then add a TestAlias to your local config.  I'll need to add one to the integration test server too.

You might consider that to be overkill though as what it will be testing mostly is the local (or whichever you alias) remote - it does make it the same as all the other backends though.

If you don't fancy that, then I think your tests should at least call NewFs - if you want to see an example of how to set that machinery up then look in [the http tests](https://github.com/ncw/rclone/blob/master/backend/http/http_internal_test.go)

**Comment by B4dM4n** [comment]
> The least effort thing to do would be to add the alias remote to gen_tests.go, run make gen_tests then add a TestAlias to your local config. I'll need to add one to the integration test server too.

I tried to use the generated tests, but they fail because alias does not have a `Object` definition:
```
# github.com/ncw/rclone/backend/alias_test
./alias_test.go:16:34: undefined: alias.Object
FAIL	github.com/ncw/rclone/backend/alias [build failed]
```
This is the failing part:
```go
func TestSetup(t *testing.T) {
	fstests.NilObject = fs.Object((*alias.Object)(nil))
	fstests.RemoteName = "TestAlias:"
}
```
Should we make the `NilObject` optional in gen_tests?
It is not used except one string compare.

> If you don't fancy that, then I think your tests should at least call NewFs - if you want to see an example of how to set that machinery up then look in the http tests

The http tests were a good start for creating the tests.
They are not very extensive, but they cover all corner cases in NewFS.

**Comment by ncw** [comment]
I think your tests look good now and are better than the gen_tests version.

I'll squash and merge...
